CONTRIBUTING.md

```diff
@@ -107,6 +107,21 @@ For MacOS users, you can install [OrbStack as a docker alternative](docs/contrib
 
 For podman, refer to [Using Podman instead of Docker](docs/contributing/podman.md)
 
+#### Install precommit hook
+
+We're configured precommit hook which does formatting and linting to catch common mistakes and improve rust code quality.
+
+To install,
+```sh
+pip install pre-commit
```

Comment:
Can we use something written in Rust? And do we really need a tool for this? It seems like it should be simple to just copy some scripts to `.git/hooks`.